cmd/cluster/encryption/encryption.go

```diff
@@ -82,6 +82,8 @@ var updateCmk = &cobra.Command{
 		authApi.GetInfo("", "")
 
 		clusterName, _ := cmd.Flags().GetString("cluster-name")
+		cloudProvider, _ := cmd.Flags().GetString("cloud-provider")
```

Comment:
This seems incorrect.

1. this is an invalid flag for the command (the command only takes an encryption spec)
2. the cloud-provider flag is usually the cloud provider of the cluster, not of the CMK (you can have GCP clusters with AWS CMK).

Reply:
You probably want to use `newCmkSpec.CloudProvider` or something equivalent in the if condition.

Reply:
Yes, you are correct, updated to fetch the cmk provider from newCmkSpec.